### DISCUSSION TITLE
NemoGuardrails Server OpenAI Compatibility

### DISCUSSION THREAD

**Discussion by dohrn** [NVIDIA-NeMo/Guardrails#444] [Q&A]
Hello, I was not able to find how to make the server compatible with OpenAI API protocol? NemoGuardrails Server exposes a different interface and Streaming seams to be impossible as well. If we lose the compatibility by integrating NemoGuardrails between Application and Model, the usefulness is very limited.

**Comment by drazvan** [comment]
@dohrn: I agree with you that we should keep compatibility with OpenAI. Currently, we have a "soft-compatibility" for the server API, in the sense that all the fields from OpenAI have the same meaning. But indeed, you can't point the OpenAI client to an guardrails server endpoint. This is also related to #451. Perhaps the server should have a default config (e.g., configurable when starting the server) and if a call is made without any guardrails-specific fields, it will use the default configuration, BUT, replace the main model with the one specified in the body. We'll try to get this in for 0.10.0 or 0.11.0.
